zip-bolt-support.rst

```diff
@@ -0,0 +1,334 @@
+::
+
+  ZIP: XXX
+  Title: Add support for Blind Off-chain Lightweight Transactions (Bolt) protocol
+  Authors: J. Ayo Akinyele <ayo@boltlabs.io>
+           Colleen Swanson <swan@boltlabs.io>
+  Credits: Ian Miers <imiers@z.cash>
+           Matthew Green <mgreen@z.cash>
+  Category: Consensus
+  Created: 2019-07-15
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST" and "MUST NOT" in this document are to be interpreted as described in RFC 2119. [#RFC2119]_
+
+Abstract
+========
+
+This proposal specifies three possible approaches for integrating the blind off-chain lightweight transaction (Bolt) protocol [#bolt-paper]_ into Zcash.
+
+Motivation
+==========
+
+Layer 2 protocols like Lightning enable scalable payments for Bitcoin but lack the mechanisms to provide strong privacy guarantees on the payment network. Zcash offers private transactions but currently lacks features that would enable a lightning-style payment channel. This proposal specifies an integration of the Bolt privacy-preserving Lightning protocol on top of Zcash [#bolt-paper]_.
+
+Specification
+=============
+
+This specification details an initial approach to integrating the features of Bolt into Zcash in a future network upgrade and depends on the WTP ZIP [#wtp-programs]_ that introduces Whitelisted Transparent Programs (WTPs). Our prototype implementation can be found here: [#BoltWTP]_.
+
+1. General requirements for Bolt protocol
+--------------------------
+
+Bolt private payment channels require the following capabilities to provide anonymity properties for users on a payment network:
```

Comment:
Explain the different roles of customer and merchant. (I believe that the customer is supposed to be anonymous to the merchant but not vice versa, and this is reflected in key lifetimes for example.)